x/emissions/keeper/query_server.go

```diff
@@ -291,7 +291,8 @@ func (qs queryServer) GetAllExistingInferenceRequests(ctx context.Context, req *
 		if err != nil {
 			return nil, err
 		}
-		ret = append(ret, &state.InferenceRequestAndDemandLeft{InferenceRequest: &inferenceRequest, DemandLeft: demandLeft})
+		inferenceRequestCopy := inferenceRequest
```

Comment:
is this enough, do we need to maybe do a deep copy?

Reply:
This is enough, just creating a transient new var. 
This has been tested locally